qt/python/mantidqtinterfaces/mantidqtinterfaces/Muon/GUI/Common/pairing_table_widget/pairing_table_widget_view.py

```diff
@@ -116,84 +126,128 @@ def set_up_table(self):
-                self.pairing_table.setItem(row_position, i, pair_name_widget)
-                item.setFlags(QtCore.Qt.ItemIsEnabled)
-                item.setFlags(QtCore.Qt.ItemIsSelectable)
-            if pair_columns[i] == "group_1":
-                group1_selector_widget = self._group_selection_cell_widget()
-                # ensure changing the selection sends an update signal
-                group1_selector_widget.currentIndexChanged.connect(lambda: self.on_cell_changed(row_position, 2))
-                index = self.get_index_of_text(group1_selector_widget, entry)
-                group1_selector_widget.setCurrentIndex(index)
-                self.pairing_table.setCellWidget(row_position, i, group1_selector_widget)
-            if pair_columns[i] == "group_2":
-                group2_selector_widget = self._group_selection_cell_widget()
-                # ensure changing the selection sends an update signal
-                group2_selector_widget.currentIndexChanged.connect(lambda: self.on_cell_changed(row_position, 3))
-                index = self.get_index_of_text(group2_selector_widget, entry)
-                group2_selector_widget.setCurrentIndex(index)
-                self.pairing_table.setCellWidget(row_position, i, group2_selector_widget)
-            if pair_columns[i] == "alpha":
-                alpha_widget = table_utils.ValidatedTableItem(self._validate_alpha)
-                alpha_widget.setText(entry)
-                self.pairing_table.setItem(row_position, i, alpha_widget)
-            if pair_columns[i] == "to_analyse":
-                if entry:
-                    item.setCheckState(QtCore.Qt.Checked)
-                else:
-                    item.setCheckState(QtCore.Qt.Unchecked)
-            self.pairing_table.setItem(row_position, i, item)
-        # guess alpha button
-        guess_alpha_widget = self._guess_alpha_button()
+        return row_position
+
+    def set_item_in_table(self, row, col, item, flags=None):
+        # Set item flags if provided
+        if flags:
+            item.setFlags(flags)
+
+        self.pairing_table.setItem(row, col, item)
+        return item
+
+    def create_table_item(self, entry, color, tooltip="", alpha=127):
+        item = QtWidgets.QTableWidgetItem(entry)
+        q_brush = self.create_brush_with_color(color, alpha)  # Using the method for creating brush
+        item.setBackground(q_brush)
+        item.setToolTip(tooltip)
+        return item
+
+    def create_brush_with_color(self, color, alpha=127):
+        q_color = QtGui.QColor(*color, alpha=alpha)
+        q_brush = QtGui.QBrush(q_color)
+        return q_brush
+
+    def set_widget_in_table(self, row, col, widget):
+        self.pairing_table.setCellWidget(row, col, widget)
+
+    def set_checkbox_in_table(self, item, checked):
+        if checked:
+            item.setCheckState(QtCore.Qt.Checked)
+        else:
+            item.setCheckState(QtCore.Qt.Unchecked)
+
```

Comment:
This seems like it could use `setChecked` and use the `checked` boolean directly.